library/std/math.qs

```diff
@@ -479,9 +587,67 @@ namespace Microsoft.Quantum.Math {
             set (t1, t2) = (t2, t1 - quotient * t2);
         }
 
-        return (r2 == 0 and AbsI(s2) <= denominatorBound)
-                ? (-t2 * signB, s2 * signA)
-                | (-t1 * signB, s1 * signA);
+        if r2 == 0 and AbsI(s2) <= denominatorBound {
+            (-t2 * signB, s2 * signA)
+        } else {
+            (-t1 * signB, s1 * signA)
+        }
+    }
+
+    /// # Summary
+    /// Finds the continued fraction convergent closest to `fraction`
+    /// with the denominator less or equal to `denominatorBound`
+    /// Using process similar to this: https://nrich.maths.org/1397
+    function ContinuedFractionConvergentL(
+        fraction : (BigInt, BigInt),
+        denominatorBound : BigInt) : (BigInt, BigInt) {
+        Fact(denominatorBound > 0L, "Denominator bound must be positive");
+
+        let (a, b) = fraction;
+        let signA = IntAsBigInt(SignL(a));
```

Comment:
Why is this necessary. a and b are already BigInt. Does SignL not return a BigInt?

Reply:
SignL returns Int, not BigInt. We may consider changing it, but IntAsBigInt is required until then.